RESTserver/src/test/java/com/softserve/if072/restservice/controller/RegistrationControllerTest.java

```diff
@@ -0,0 +1,99 @@
+
+/**
+ * Contains tests for RegistrationController methods
+ *
+ * @author Igor Parada
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class RegistrationControllerTest {
+
+    @Mock
+    private UserService userService;
+    private RegistrationController registrationController;
+    private MockMvc mockMvc;
+    private User user;
+    private ObjectMapper mapper;
+
+    @Before
+    public void setup() throws ClassNotFoundException, NoSuchMethodException {
+        registrationController = new RegistrationController(userService);
+        mockMvc = standaloneSetup(registrationController).build();
+        mapper =  new ObjectMapper();
+
+
+        user = new User();
+        user.setEmail("test@user.com");
+        user.setPassword("test");
+        user.setName("testname");
+        user.setRole(Role.ROLE_REGULAR);
+
+        when(userService.getByUsername("test@user.com")).thenReturn(user);
+    }
+
+    @Test
+    public void testPerformRegistrationFailWhenRequestBodyIsEmpty() throws Exception {
+        mockMvc.perform(post("/register/").contentType(
+                MediaType.APPLICATION_JSON).content("")).andExpect(
+                status().is4xxClientError());
+        verifyZeroInteractions(userService);
+    }
+
+    @Test
+    public void performRegistration_ShouldFailWhenUserAlreadyExist() throws Exception {
+
+        mockMvc.perform(post("/register/").contentType(
+                MediaType.APPLICATION_JSON).content(mapper.writeValueAsString(user))).andExpect(
+                status().is4xxClientError());
+        verify(userService, times(0)).insert(any());
+    }
+
+    @Test
+    public void performRegistration_ShouldReturnStatus200WhenSuccessful() throws Exception {
+        User newUser = new User();
+        newUser.setEmail("new@user.com");
+        newUser.setPassword("newpassword");
+        newUser.setName("newname");
+        newUser.setRole(Role.ROLE_REGULAR);
+
+        mockMvc.perform(post("/register/").contentType(
+                MediaType.APPLICATION_JSON).content(mapper.writeValueAsString(newUser))).andExpect(
+                status().isOk());
+        verify(userService, times(1)).insert(any());
+    }
+
+    @Test
+    public void findUsername_ShouldReturnEmptyStringWhenUsernameNotExists() throws Exception {
+        mockMvc.perform(post("/register/findUsername").content("another@user.com"))
+                .andExpect(status().isOk())
+                .andExpect(content().string(""));
+        verify(userService, times(1)).getByUsername("another@user.com");
+    }
+
+    @Test
+    public void findUsername_ShouldReturnUsernameWhenUsernameAlreadyExists() throws Exception {
+        mockMvc.perform(post("/register/findUsername").content("test@user.com"))
+                .andExpect(status().isOk())
+                .andExpect(content().string("test@user.com"));
+        verify(userService, times(1)).getByUsername("test@user.com");
```

Comment:
can be
`verify(userService).getByUsername("test@user.com");`

When you expect 1 interection with the mock, them times(1) is optional